ruma-federation-api/src/membership/create_invite/v1.rs

```diff
@@ -0,0 +1,146 @@
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The room ID that the user is being invited to.
+        #[ruma_api(path)]
+        room_id: RoomId,
+
+        /// The event ID for the invite event, generated by the inviting server.
+        #[ruma_api(path)]
+        event_id: EventId,
+
+        /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+        sender: UserId,
+
+        /// The name of the inviting homeserver.
+        origin: Box<ServerName>,
+
+        /// A timestamp added by the inviting homeserver.
+        origin_server_ts: UInt,
+
+        /// The value `m.room.member`.
+        #[serde(rename = "type")]
+        kind: EventType,
+
+        /// The user ID of the invited member.
+        state_key: UserId,
+
+        /// The content of the event.
+        content: MemberEventContent,
+
+        /// Information included alongside the event that is not signed.
+        unsigned: UnsignedEventContent,
+    }
+
+    response: {
+        /// The response invite event
+        #[ruma_api(body)]
+        #[serde(with = "crate::serde::invite_response")]
+        invite_response: InviteEvent,
+    }
+}
+
+/// Information included alongside an event that is not signed.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UnsignedEventContent {
+    /// An optional list of simplified events to help the receiver of the invite identify the room.
+    /// The recommended events to include are the join rules, canonical alias, avatar, and name of
+    /// the room.
+    invite_room_state: Vec<StrippedState>,
+}
+
+/// Initial set of fields of `Request`.
+pub struct RequestInit {
+    /// The room ID that the user is being invited to.
+    room_id: RoomId,
+
+    /// The event ID for the invite event, generated by the inviting server.
+    event_id: EventId,
+
+    /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+    sender: UserId,
+
+    /// The name of the inviting homeserver.
+    origin: Box<ServerName>,
+
+    /// A timestamp added by the inviting homeserver.
+    origin_server_ts: UInt,
+
+    /// The user ID of the invited member.
+    state_key: UserId,
+
+    /// The content of the event.
+    content: MemberEventContent,
+
+    /// Information included alongside the event that is not signed.
+    unsigned: UnsignedEventContent,
+}
+
+/// Initial set of fields for `Response`.
+pub struct ResponseInit {
+    /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+    sender: UserId,
+
+    /// The name of the inviting homeserver.
+    origin: Box<ServerName>,
+
+    /// A timestamp added by the inviting homeserver.
+    origin_server_ts: UInt,
+
+    /// The user ID of the invited member.
+    state_key: UserId,
+
+    /// The content of the event. Must include a `membership` of invite.
+    content: MemberEventContent,
+}
+
+impl Request {
```

Comment:
Can you make this a `From` conversion instead?

```suggestion
impl From<RequestInit> for Request {
```